src/components/checkBoxSelection/checkBoxSelection.js

```diff
@@ -1,47 +1,106 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import './checkBoxSelection.scss';
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import "./checkBoxSelection.scss";
+import { connect } from "react-redux";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class CheckBoxSelection extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: false
-        };
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected
+    };
+  }
+
+  convertSvgToDataUrl = file => {
+    let reader = new FileReader();
+    reader.onload = result => {
+      this.props.svgSettingList.push({
+        name: file.name,
+        dataurl: result.target.result
+      });
+    };
+    if (file) {
+      reader.readAsDataURL(file);
     }
+  };
 
-    componentWillMount() {
-	 this.setState({ isClicked: this.props.isSelected }); 
+  changeColourAndTick = () => {
+    let isFound = false;
+    let index = 0;
+    this.setState({ isClicked: !this.state.isClicked });
+    for (var key in this.props.svgSettingList) {
+      if (
+        this.props.svgSettingList[key].name === this.props.file.name
+      ) {
+        isFound = true;
+        index = key;
+        break;
+      }
     }
+    if (!isFound) {
+      this.convertSvgToDataUrl(this.props.file);
+    } else {
+      this.props.svgSettingList.splice(index, 1);
```

Comment:
splice changes the origional array, mutability is bad practice in react

Reply:
Could you please also assist me in suggesting solutions or recommendations to use to avoid mutability ?

Reply:
nvm

Reply:
cool